### PR TITLE
Fix Smoothed Aggregation corner cases and test

### PR DESCRIPTION
**Pull request by termi-official**

### CODE REVIEW

File: test/sa_tests.jl [hunk 1]
```diff
@@ -139,6 +134,58 @@ function stand_agg(C, ϵ=0)
...
+
+
+# Corner case tests for StandardAggregation
+function test_standard_aggregation_corner_cases()
+
+    # Off-by-one in aggregate_size: a 4-node chain whose strength matrix has no
+    # diagonal entries should form 2 aggregates of size 2 when min_aggregate_size=2.
+    # The buggy code (aggregate_size starts at 0) rejects the first candidate because
+    # it only counts the 1 neighbour and misses the seed, collapsing everything into
+    # one aggregate instead.
```
Comment (Abdelrahman912) [inline test/sa_tests.jl]: I guess this should be fixed because `min_aggregate_size` is already removed from `StandardAggregation`

**Review by Abdelrahman912 — APPROVED** [review]
